orca-echo/src/main/groovy/com/netflix/spinnaker/orca/echo/tasks/CreateJiraIssueTask.java

```diff
@@ -0,0 +1,284 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.echo.tasks;
+
+import com.fasterxml.jackson.annotation.*;
+import com.google.common.collect.ImmutableMap;
+import com.netflix.spinnaker.orca.ExecutionStatus;
+import com.netflix.spinnaker.orca.RetryableTask;
+import com.netflix.spinnaker.orca.TaskResult;
+import com.netflix.spinnaker.orca.echo.EchoService;
+import com.netflix.spinnaker.orca.pipeline.model.Execution;
+import com.netflix.spinnaker.orca.pipeline.model.Stage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.Nonnull;
+import javax.validation.ConstraintViolation;
+import javax.validation.Validation;
+import javax.validation.constraints.NotNull;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+@Component
+public class CreateJiraIssueTask implements RetryableTask {
+  private static Logger LOGGER = LoggerFactory.getLogger(CreateJiraIssueTask.class);
+  private final EchoService echoService;
+
+  @Autowired
+  public CreateJiraIssueTask(EchoService echoService) {
+    this.echoService = echoService;
+  }
+
+  @Override
+  public long getBackoffPeriod() {
+    return TimeUnit.SECONDS.toMillis(30);
+  }
+
+  @Override
+  public long getTimeout() {
+    return TimeUnit.MINUTES.toMillis(5);
+  }
+
+  @Nonnull
+  @Override
+  public TaskResult execute(@Nonnull Stage stage) {
+    CreateIssueContext createIssueContext = stage.mapTo(CreateIssueContext.class);
+    validateInputs(createIssueContext);
+    EchoService.Notification notification = new EchoService.Notification();
+    EchoService.Notification.Source source = new EchoService.Notification.Source();
+    String user = getAuthenticatedUser(stage);
+    source.setUser(user);
+    createIssueContext.getFields().setReporter(new CreateIssueContext.Fields.Reporter().withName(user));
+    notification.setTo(Collections.singletonList(source.getUser()));
+    notification.setSource(source);
+    notification.setNotificationType(EchoService.Notification.Type.JIRA);
+    notification.setAdditionalContext(ImmutableMap.of("issueContext", createIssueContext));
+    echoService.create(notification);
+    LOGGER.info("Created Jira Issue {} for {}", createIssueContext, source.getUser());
+    return new TaskResult(ExecutionStatus.SUCCEEDED);
+  }
```

Comment:
This method could really use some vertical whitespace. As-is, it's hard to read even though there's not much going on.